src/immer.d.ts

```diff
@@ -12,27 +12,17 @@ type AtomicObject =
     | Number
     | String
 
-type ArrayMethod = Exclude<keyof [], number>
-type Indices<T> = Exclude<keyof T, ArrayMethod>
-
-export type DraftArray<T extends ReadonlyArray<any>> = Array<
-    {[P in Indices<T>]: Draft<T[P]>}[Indices<T>]
->
-
-export type DraftTuple<T extends ReadonlyArray<any>> = {
-    [P in keyof T]: P extends Indices<T> ? Draft<T[P]> : never
-}
-
-export type Draft<T> = T extends never[]
+export type Draft<T> = T extends AtomicObject
     ? T
-    : T extends ReadonlyArray<any>
-    ? T[number][] extends T
-        ? DraftArray<T>
-        : DraftTuple<T>
-    : T extends AtomicObject
+    : T extends object
+    ? {-readonly [K in keyof T]: Draft<T[K]>}
+    : T // mostly: unknown & any
```

Comment:
And also primitives like `number | string | boolean` 😛 Or am I misunderstanding this comment?

Reply:
eh yes, had a `never` first there, and `AtomicObject` at the start, will remove the comment